interfaces/builtin/optical_drive.go

```diff
@@ -0,0 +1,43 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const opticalDriveConnectedPlugAppArmor = `
+/dev/sr0 r,
+`
+
+const opticalDriveConnectedPlugSecComp = `
+ioctl
+`
+
+// NewOpticalDriveInterface returns a new "optical-drive" interface.
+func NewOpticalDriveInterface() interfaces.Interface {
+	return &commonInterface{
+		name: "optical-drive",
+		connectedPlugAppArmor: opticalDriveConnectedPlugAppArmor,
+		connectedPlugSecComp:  opticalDriveConnectedPlugSecComp,
+		reservedForOS:         true,
+		autoConnect:           true,
```

Comment:
I'm conflicted on autoConnect. My gut says manual connect and auto-approve in the store since a daemon would be able to poll these devices and steal data. If we are autoConnecting I would feel we would need to set to manual approve in the store which would be difficult for developers....